### PR TITLE
Grid-update

### PR DESCRIPTION
**Pull request by adriantomic**
- Added centering classes for columns
- Added push & pull classes for columns
- Changed row to work with push/pull classes.

Simple preview/kitchensink can be found here: https://www.dropbox.com/s/tacd10d253ohq8p/dbrmn-scss-framework_kitchensink.zip?dl=0

### CODE REVIEW

File: scss/modules/_grid.scss [hunk 1]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
   .container-fluid { width: 100%; }
 
   .container {
+    margin: 0 auto;
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Don't see a reason to set vertical margin here. If it's just a shorthand to replace what we had I think it has unnecessary side-effects.

File: scss/modules/_grid.scss [hunk 2]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
+      content: " ";
+      display: table;
+    }
+
+    &:after,
+    & &:after {
+      clear: both;
+    }
+
+    & & {
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: If this is to target specific nested rows I think it need some comment to explain that use-case.

File: scss/modules/_grid.scss [hunk 3]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
-    @include clearfix-overflow-visible;
-    margin-right: -$grid-col-padding;
-    margin-left: -$grid-col-padding;
+    margin-right: auto;
+    margin-left: auto;
+    
+    &:before,
+    &:after,
+    & &:before,
+    & &:after  {
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Since &:before and & &:before adds the same styles the double & selectors shouldn't be needed, or am I missing something?

File: scss/modules/_grid.scss [hunk 4]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
+      content: " ";
+      display: table;
+    }
+
+    &:after,
+    & &:after {
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Same as above.

File: scss/modules/_grid.scss [hunk 5]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
-    margin-right: -$grid-col-padding;
-    margin-left: -$grid-col-padding;
+    margin-right: auto;
+    margin-left: auto;
+    
+    &:before,
+    &:after,
+    & &:before,
+    & &:after  {
+      content: " ";
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Change to use clearfix in framework.

File: scss/modules/_grid.scss [hunk 6]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
-  .col-opposite { float: right; }
-
   @for $i from 1 through $grid-cols {
     .col-#{$i},
     .col-sm-#{$i} {
       float: left;
       width: (100% / $grid-cols) * $i;
       padding-right: $grid-col-padding;
       padding-left: $grid-col-padding;
+      position: relative;
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Should all col classes have position relative? Then move all these to .col {}

File: scss/modules/_grid.scss [hunk 7]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
     }
   }
-}
+  @for $i from 1 through $grid-cols {
+    .col-push-#{$i},
+    .col-sm-push-#{$i} {
+      float: left;
+      position: relative;
+      left: (100% / $grid-cols) * $i;
+    }
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line-break

File: scss/modules/_grid.scss [hunk 8]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
+    .col-pull-#{$i},
+    .col-sm-pull-#{$i} {
+      position: relative;
+      right: (100% / $grid-cols) * $i;
+      left: auto;
+      float: right;
+
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Remove white space.

File: scss/modules/_grid.scss [hunk 9]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
+    }
+    .col-pull-#{$i},
+    .col-sm-pull-#{$i} {
+      position: relative;
+      right: (100% / $grid-cols) * $i;
+      left: auto;
+      float: right;
+      
+    }
+  }
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line break.

File: scss/modules/_grid.scss [hunk 10]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
+  .col-center,
+  .col-sm-center {
+    float: none;
+    margin: 0 auto;
+  }
 
-/* _grid small */
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Why is this removed?

File: scss/modules/_grid.scss [hunk 11]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
-@include layout-small {
-  .hide-sm { display: none; }
+  .col-opposite,
+  .col-sm-opposite { float: right;  }
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Remove double white space.

File: scss/modules/_grid.scss [hunk 12]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
     padding-left: $base-unit * 2;
   }
 
+  .col-md-opposite { float: right;  }
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Remove double white space and add line break.

File: scss/modules/_grid.scss [hunk 13]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
+
   @for $i from 1 through $grid-cols {
     .col-md-#{$i} {
       float: left;
       width: (100% / $grid-cols) * $i;
+      padding-right: $grid-col-padding;
+      padding-left: $grid-col-padding;
+      position: relative;
+    }
+  }
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line break.

File: scss/modules/_grid.scss [hunk 14]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
+  @for $i from 1 through $grid-cols {
+    .col-push-#{$i},
+    .col-md-push-#{$i} {
+      float: left;
+      position: relative;
+      left: (100% / $grid-cols) * $i;
+    }
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line break.

File: scss/modules/_grid.scss [hunk 15]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
+    .col-pull-#{$i},
+    .col-md-pull-#{$i} {
+      position: relative;
+      right: (100% / $grid-cols) * $i;
+      left: auto;
+      float: right;
+
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Remove blank line.

File: scss/modules/_grid.scss [hunk 16]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
+  .col-md-lg-center {
+    margin-left: auto;
+    margin-right: auto;
+  }
 }
 
 /* _grid large */
 @include layout-large {
   .hide-lg { display: none; }
+  .col-lg-opposite { float: right;  }
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line break.

File: scss/modules/_grid.scss [hunk 17]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
+  .col-lg-center {
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line break.

File: scss/modules/_grid.scss [hunk 18]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
     .col-lg-#{$i} {
       float: left;
       width: (100% / $grid-cols) * $i;
+      padding-right: $grid-col-padding;
+      padding-left: $grid-col-padding;
+      position: relative;
     }
   }
-}
+  @for $i from 1 through $grid-cols {
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line break.

File: scss/modules/_grid.scss [hunk 19]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
+    .col-push-#{$i},
+    .col-lg-push-#{$i} {
+      float: left;
+      position: relative;
+      left: (100% / $grid-cols) * $i;
+    }
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line break.

File: scss/modules/_grid.scss [hunk 20]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
+    .col-pull-#{$i},
+    .col-lg-pull-#{$i} {
+      position: relative;
+      right: (100% / $grid-cols) * $i;
+      left: auto;
+      float: right;
+
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Remove blank line.

File: scss/modules/_grid.scss [hunk 21]
```diff
@@ -52,27 +87,80 @@ $grid-col-padding: 15px;
...
+      float: right;
+      
     }
   }
 }
 
 /* _grid medium-and-above */
 @include layout-medium-and-above {
   .hide-md-lg { display: none; }
+  .col-md-lg-center {
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Add line break.

File: scss/modules/_grid.scss [hunk 22]
```diff
@@ -9,38 +9,73 @@ $grid-col-padding: 15px;
...
       width: (100% / $grid-cols) * $i;
       padding-right: $grid-col-padding;
       padding-left: $grid-col-padding;
+      position: relative;
     }
   }
-}
+  @for $i from 1 through $grid-cols {
+    .col-push-#{$i},
+    .col-sm-push-#{$i} {
```
Comment (larsericsson) [inline scss/modules/_grid.scss]: Put sm classes in layout-small. Looks like this is an old error, but maybe a good time to set it straight :)

File: scss/modules/_grid.scss (hunk 1 above)
Comment (adriantomic) [inline scss/modules/_grid.scss]: Removed vertical margin!

File: scss/modules/_grid.scss (hunk 2 above)
Comment (adriantomic) [inline scss/modules/_grid.scss]: Yes, I'm targeting rows that are nested in other rows. Will add comments to explain.

File: scss/modules/_grid.scss (hunk 3 above)
Comment (adriantomic) [inline scss/modules/_grid.scss]: You're right. I did some cleanup.

File: scss/modules/_grid.scss (hunk 5 above)
Comment (adriantomic) [inline scss/modules/_grid.scss]: Done!

File: scss/modules/_grid.scss (hunk 6 above)
Comment (adriantomic) [inline scss/modules/_grid.scss]: Maybe better to target .col with [class^="col-"] {} ? Sometimes .col is omitted when declaring columns to elements in projects. I'd like to keep it as it is. Eg, with position relative at two places in the code.

File: scss/modules/_grid.scss (hunk 10 above)
Comment (adriantomic) [inline scss/modules/_grid.scss]: Sorry! Put it back!

File: scss/modules/_grid.scss (hunk 22 above)
Comment (adriantomic) [inline scss/modules/_grid.scss]: Fixed

**Comment by jacobgunnarsson** [comment]
Closing this due to removal of _grid.scss alltogether.
